apps/meteor/app/livechat/imports/server/rest/appearance.ts

```diff
@@ -52,8 +52,39 @@ API.v1.addRoute(
 				throw new Error('invalid-setting');
 			}
 
+			const dbSettings = await Settings.find({ _id: { $in: validSettingList } }, { projection: { _id: 1, value: 1, type: 1 } }).toArray();
+
+			const settingsToSave: typeof this.bodyParams = [];
+			dbSettings.forEach((dbSetting) => {
```

Comment:
I believe we could use the map loop here instead of creating a new one 👀

Reply:
You know, I always forget maps exist